src/beast/beast/container/vectum.h

```diff
@@ -0,0 +1,642 @@
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef BEAST_CONTAINER_VECTUM_H_INCLUDED
+#define BEAST_CONTAINER_VECTUM_H_INCLUDED
+
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+#include <iterator>
+#include <memory>
+#include <new>
+#include <beast/cxx14/type_traits.h> // <type_traits>
+#include <utility>
+
+namespace beast {
+namespace nudb {
+namespace detail {
+
+template <class>
+class vectum;
+
+template <class T>
+void
+swap (vectum<T>&, vectum<T>&);
+
+// Custom vector where the element size is controlled
+// at run-time during construction and can be larger than
+// the value_type.
+//
+template <class T>
+class vectum
+{
+private:
+    template <bool is_const>
+    class basic_iterator;
+    
+    std::size_t size_ = 0;
+    std::size_t capacity_ = 0;
+    std::size_t value_size_;
+    std::size_t aligned_size_;
+    std::unique_ptr<std::uint8_t[]> buf_;
+
+public:
+    using value_type = T;
+    //using allocator_type = Alloc;
+    using size_type = std::size_t;
+    using difference_type = std::ptrdiff_t;
+    using reference = value_type&;
+    using const_reference = value_type const&;
+    using pointer = value_type*;
+    using const_pointer = value_type const*;
+    using iterator = basic_iterator<false>;
+    using const_iterator = basic_iterator<true>;
+    using reverse_iterator = std::reverse_iterator<iterator>;
+    using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+
+    // Non-trivial construction is allowed because we
+    // don't support vector operations like resize().
+#ifdef _MSC_VER
+    static_assert(
+        //std::is_trivially_destructible<T>::value &&
+        std::is_trivially_copy_constructible<value_type>::value &&
+        std::is_trivially_copy_assignable<value_type>::value &&
+        std::is_trivially_move_constructible<value_type>::value &&
+        std::is_trivially_move_assignable<value_type>::value,
+            "only trivial types supported");
+#else
+    // glibc++ doesn't seem to handle this
+#endif
+
+    vectum (vectum const&) = delete;
+    vectum& operator= (vectum const&) = delete;
+
+    explicit
+    vectum (size_type value_size = sizeof(value_type));
+
+    vectum& operator= (vectum&& other);
+
+    reference
+    at (size_type n)
+    {
+        if (n >= size_)
+            throw std::out_of_range(
```

Comment:
If you're going to make these two lines (why?) then please put braces around them...